File: WdtcCore/src/main/java/org/wdt/wdtc/core/manger/DownloadSourceManger.java
package org.wdt.wdtc.core.manger;

import org.wdt.wdtc.core.download.infterface.DownloadSourceInterface;
import org.wdt.wdtc.core.download.source.BmclDownloadSource;
import org.wdt.wdtc.core.download.source.McbbsDownloadSource;
import org.wdt.wdtc.core.download.source.OfficialDownloadSource;

public class DownloadSourceManger {
    public static DownloadSourceInterface getOfficialDownloadSource() {
        return new OfficialDownloadSource();
    }

    public static DownloadSourceManger.DownloadSourceList getDownloadSourceKind() {
        return SettingManger.getSetting().getDownloadSource();
    }

    public static DownloadSourceInterface getDownloadSource() {
        DownloadSourceList source = getDownloadSourceKind();
        if (source == DownloadSourceList.OFFICIAL) {
            return new OfficialDownloadSource();
        } else if (source == DownloadSourceList.BMCLAPI) {
            return new BmclDownloadSource();
        } else {
            return new McbbsDownloadSource();
        }
    }

    public static boolean isNotOfficialDownloadSource() {
        return !isOfficialDownloadSource();
    }

    public static boolean isOfficialDownloadSource() {
        return getDownloadSourceKind() == DownloadSourceList.OFFICIAL;
    }

    public enum DownloadSourceList {
        OFFICIAL,
        BMCLAPI,
        MCBBS;

    }
}

File: WdtcCore/src/main/kotlin/org/wdt/wdtc/core/manger/DownloadSourceManger.kt
@file:JvmName("DownloadSourceManger")

package org.wdt.wdtc.core.manger

import org.wdt.wdtc.core.download.infterface.DownloadSourceInterface
import org.wdt.wdtc.core.download.source.BmclDownloadSource
import org.wdt.wdtc.core.download.source.McbbsDownloadSource
import org.wdt.wdtc.core.download.source.OfficialDownloadSource

val officialDownloadSource: DownloadSourceInterface = OfficialDownloadSource()

val currentDownloadSourceKind: DownloadSourceKind
  get() = currentSetting.downloadSource

val currentDownloadSource: DownloadSourceInterface
  get() {
	  return when (currentDownloadSourceKind) {
      DownloadSourceKind.OFFICIAL -> {
        OfficialDownloadSource()
      }

      DownloadSourceKind.BMCLAPI -> {
        BmclDownloadSource()
      }

      else -> {
        McbbsDownloadSource()
      }
    }
  }

val isOfficialDownloadSource: Boolean
	get() = currentDownloadSourceKind == DownloadSourceKind.OFFICIAL

val isNotOfficialDownloadSource: Boolean
  get() = !isOfficialDownloadSource

enum class DownloadSourceKind {
  OFFICIAL,
  BMCLAPI,
  MCBBS;
  
}
